scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -436,23 +436,32 @@ public void updateActivityDirective(
     for (final var arg : activity.arguments().entrySet()) {
       arguments.add(arg.getKey(), serializedValueP.unparse(arg.getValue()));
     }
-    final var query = """
-        mutation UpdateActivity($id: Int!, $planId: Int!, $arguments: String!, $startOffset: Int!, $goalId: Int) {
-          update_activity_directive_by_pk(pk_columns: {id: $id, plan_id: $planId}, _set: {arguments: $arguments, start_offset: $startOffset, source_scheduling_goal_id: $goalId}) {
-            affected_rows
+    String query;
+    final var variables = Json.createObjectBuilder()
+                    .add("id", instanceId.id())
+                    .add("planId", planId.id())
+                    .add("arguments", arguments.build())
+                    .add("startOffset", activity.startTimestamp().toString());
+    if(goalId != null) {
+      query = """
```

Comment:
I would expect that we can have a null value for goalId and use the same query in both cases. We just need to do a null check before calling `.id()` on the goal.

Reply:
It sounds like we might not even need a null check, as long as the json object builder accepts `.add("goalId", null)`. If so, I recommend dropping the second commit.

Side note: the `updateActivityDirective` shouldn't be called unless the scheduler is updating an activity that's already in the plan, right? So in theory, even if it's broken, none of our codepaths should trigger it. If we are hitting that codepath, that means we have another issue upstream. 🤔

Reply:
https://docs.oracle.com/javaee/7/api/javax/json/JsonObjectBuilder.html

> This class does not allow null to be used as a name or value while building the JSON object

The docs are admittedly outdated, but for some reason my IntelliJ won't let me jump to the sources -- just to decompiled bytecode. Anyway, this matches my understanding last I looked.

Reply:
Finally, these changes have been reverted. I have also commented calls to delete and update. If we end up in this branch anyway, we'll get an exception, the scheduler is not supposed to do that (yet).